src/util.rs

```diff
@@ -73,13 +76,16 @@ pub fn encrypt<RNG: RngCore + CryptoRng>(
 }
 
 /// Decrypt the passed data using the key
-pub fn decrypt(key: &[u8; 32], data: &[u8]) -> Result<Vec<u8>, AesGcmError> {
+pub fn decrypt(key: &[u8; 32], data: &[u8]) -> Result<Vec<u8>, EncryptionError> {
+    if data.len() < AES_GCM_NONCE_SIZE {
+        return Err(EncryptionError::MissingNonce);
+    }
     let nonce_vec = data[..AES_GCM_NONCE_SIZE].to_vec();
     let cipher_vec = data[AES_GCM_NONCE_SIZE..].to_vec();
```

Comment:
I think it's better to do something like this:
```rust
let Some(nonce) = data.get(..AES_GCM_NONCE_SIZE) else {
    return Err(Error::MissingNonce);
};
let Some(cipher_data) = data.get(AES_GCM_NONCE_SIZE..) else {
    return Err(Error::MissingData);
};
...
```
It does what we want and returns a more detailed error. Also, the lines that create `nonce_vec` and `cipher_vec` don't need to create vectors, but should just create slices. It's a little more efficient that way.